containers/software-containers/arcade-casa/version-6.1-6.3/Dockerfile

```diff
@@ -0,0 +1,62 @@
+FROM centos:6
+
+# xterm dependency is an extra to get the casa shell in the display
+# perl was added for casa later than 5
+
+# Override old repo info with current urls
+RUN rm /etc/yum.repos.d/CentOS-Base.repo
+ADD CentOS-Base.repo /etc/yum.repos.d/
+
+RUN yum clean all -y
+RUN yum makecache -y
+RUN yum update -y
+RUN yum install -y freetype libSM libXi libXrender libXrandr \
+	libXfixes libXcursor libXinerama fontconfig \
+        libxslt xauth xorg-x11-server-Xvfb dbus-x11 \
+	tkinter ImageMagick-c++ xterm perl autoconf python-sphinx graphviz xz
+
+# setup all required env variables
+ARG CASA_RELEASE
+ENV CASA_RELEASE=${CASA_RELEASE}
+ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/casa/bin
+
+# unpack and move casa databundle to container
+ADD ${CASA_RELEASE}.tar.xz /opt/
+
+# ADD does not untar xz automatically
+# RUN unxz /opt/${CASA_RELEASE}.tar.xz
+# RUN tar -xf /opt/${CASA_RELEASE}.tar
+
+# chown because the untarred casa has wrong owner/group
+RUN chown -R root:root /opt/${CASA_RELEASE%%-py*} && ln -s /opt/${CASA_RELEASE%%-py*} /opt/casa
+
+RUN yum install -y sssd-client acl
+
+# add the admit enhancement (issue #25)
+RUN yum install -y tcsh
+RUN mkdir /opt/admit
+ADD admit /opt/admit
+RUN cd /opt/admit && \
+    autoconf && ./configure --with-casa-root=/opt/${CASA_RELEASE%%-py*}
+
+# Allow runtime symlink creation to the casa-data-repository
+#RUN rm -rf /opt/${CASA_RELEASE%%-py*}/data
+#update for CASA6+, new directory for the leapsecond data
+RUN rm -rf /opt/${CASA_RELEASE%%-py*}/lib
+RUN chmod 777 /opt/${CASA_RELEASE%%-py*}
```

Comment:
Where is the link to casa-data-repository? How do the leapsecond data get updated?

Reply:
Hi Alinga,

How very strange... that entire directory shouldn't be there... it was something I had temporarily worked on before you implemented the leapsecond correction and I was sure I had deleted again.  (Local git issues means I cannot check how things look locally right now, unfortunately... I am going to need to delete my entire repo locally and start/clone again once the casa containers go through acceptance).  You can tell from the directory names that all of the casa containers intended to be built by this Dockerfile (in 6.1-6.3) are actually built also in the 6.1-6.4 directory which has all of the recent updates, etc.  The time stamp for the directory also shows it wasn't looked at for much longer.  My apologies for this mistaken upload.  Are you able to delete the 6.1-6.3 directory entirely and then accept the rest of the changes?

Reply:
Also, many thanks for looking through this so carefully this morning - I really appreciate it!

Reply:
Hi Helen,
I am afraid I cannot delete the 6.1-6.3 directory. You can try deleting the directory locally and then push the change.
You are very welcome!

Reply:
Hi Alinga,

Hmmm... I'm not sure what to do then.  My local github got into this really weird state with some documentation stuff I've been working on since pushing the casa_container stuff.  It's on a separate branch, but it's somehow gotten to a stage where it won't let me update my local repo, nor push my documentation updates to my personal github... and it also will not let me change branches to even get to the casa_container branch.  What a frustrating mess.

Would the best solution be for me to copy over all of the CASA container updates to some other directory (and same with the documentation), then delete everything in my local github, re-clone, re-copy back the casa container stuff, make sure the 6.1-6.3 directory is not there, and initiate a pull request again?  Or is there another simpler solution I'm not seeing?

thanks,
Helen

Reply:
Hi Helen,

If you:
- create a temp directory somewhere, 
- clone your gitHub repo inside the temp directory, 
- checkout the casa_container branch, 
you would have all the changes in this PR. 

Then you could:
- delete the version-6.1-6.3 directory,
- git rm the version-6.1-6.3 directory,
- git commit and push the change. 
Afterwards you could delete the temp directory and all subdirectories.

Hope the above helps.

Alinga

Reply:
Thanks for the advice, Alinga.  Unfortunately I just keep running into more issues and don't seem to be getting anywhere.  I tried your advice, but 'git clone github.com:helenkirk/skaha.git' transferred over only the master branch and not the casa_containers one.  I've tried forcing a clone of just the casa_containers branch using the --single-branch switch but get an error message about a lack of access to the branch (either permissions or does not exist).  I've tried just going ahead and creating a new casa_containers branch locally, then copying in all of the updated container software, doing git add and git commit and then pushing.  But that gives me an error message about my current branch being behind the remote counterpart.  Argh!  The only solution I'm seeing at this point is to make absolutely certain I have a copy of all of my changes somewhere else locally, then delete both my on-laptop repo and my on-github repo and start afresh on both places.  A slightly scary thought :(

Reply:
I have a typo above. It should have been "casa_containers" instead of "casa_container".

I am able to clone your repo and checkout the casa_containers branch.

```
alinga@NRC-023053 temp % git clone https://github.com/helenkirk/skaha.git
Cloning into 'skaha'...
remote: Enumerating objects: 9217, done.
remote: Counting objects: 100% (1471/1471), done.
remote: Compressing objects: 100% (642/642), done.
remote: Total 9217 (delta 777), reused 1412 (delta 755), pack-reused 7746
Receiving objects: 100% (9217/9217), 136.96 MiB | 2.21 MiB/s, done.
Resolving deltas: 100% (5261/5261), done.
alinga@NRC-023053 temp % cd skaha 
alinga@NRC-023053 skaha % git remote -v
origin	https://github.com/helenkirk/skaha.git (fetch)
origin	https://github.com/helenkirk/skaha.git (push)
alinga@NRC-023053 skaha % git checkout casa_containers
branch 'casa_containers' set up to track 'origin/casa_containers'.
Switched to a new branch 'casa_containers'
alinga@NRC-023053 skaha % git status
On branch casa_containers
Your branch is up to date with 'origin/casa_containers'.

nothing to commit, working tree clean
```

Reply:
Here are the available branches:

```
alinga@NRC-023053 skaha % pwd
/Users/alinga/work/git/temp/skaha
alinga@NRC-023053 skaha % git branch -a
* casa_containers
  master
  remotes/origin/HEAD -> origin/master
  remotes/origin/add_casa_containers
  remotes/origin/casa_containers
  remotes/origin/documentation
  remotes/origin/find_orb
  remotes/origin/lsst
  remotes/origin/master
  remotes/origin/ossos
  remotes/origin/starlink
  remotes/origin/starlink_jjk
```